packages/core/src/utils/client.ts

```diff
@@ -101,26 +115,45 @@ export class Client {
       status: x.data?.activeKey ? StatusEnum.RECONNECTING : StatusEnum.CONNECTING,
     }));
 
-
     let isConnected = false;
     for (const connector of this.connectors || []) {
       const isConnectedWithConnector = await connector?.isConnected();
-
       if (isConnectedWithConnector) {
-        await this.connector?.connect();
-        const publicKey = await connector?.getActivePublicKey();
-        this.setState((x: StateParams) => ({
-          ...x,
-          status: StatusEnum.CONNECTED,
-          connector,
-          data: {
-            ...x.data,
-            activeKey: publicKey,
-          },
-        }));
-        isConnected = true;
-
-        break;
+        let publicKey: string | undefined;
+        try {
```

Comment:
I see this block being repeated in multiple places:
```
try {
  publicKey = await this.connector?.getActivePublicKey();
} catch (err) {
  console.error(err);
  publicKey = undefined;
}
```

It'd be better to move this snippet into small function, something like:
```
this.getPublicKey = () => {
...// insert here
}
```
or even better, use:
```
get publicKey () {
...
}
```